### DISCUSSION TITLE
Field restrictions

### DISCUSSION THREAD

**Discussion by bloor** [pocketbase/pocketbase#1041] [Q&A]
![image](https://user-images.githubusercontent.com/1420653/202252991-29bbda6c-4117-4c73-aff2-c9465760f411.png)

If field is empty/null, I get "Min max restrictions" error.
Correct me if I`m wrong, but shouldn't the field be allowed to be NULL in this scenario ?
I have a specific case where I want the "admin" to not be "forced" to fill that field, but in case he wants to input, to restrict the input accordingly.

**Comment by ganigeorgiev** [comment]
With the exception of the `json` type field, PocketBase doesn't store `null` for field values and uses zero-defaults (for number that is zero). Or in other words, you'll never have `null` as value for a number field and missing numbers will be always `0` and will trigger your validation.

**Reply by ganigeorgiev** [reply]
To avoid confusion, I'll consider to disable/hide the "Nonempty" field when "Min" is set (for `text` and `number` field types).

**Reply by ganigeorgiev** [reply]
@bloor Actually I stand corrected. After reviewing the validation logic in more details the current behavior is a bug.
When "Required/Nonempty" is NOT checked the `min` validator should NOT get triggered. Currently it conflicts with the "Nonempty" tooltip description and is not consistent with what we do for the `text` field type. I'll submit a fix a little bit later and it will be available with the final v0.8.0 release.
